src/Model/ROI.py

```diff
@@ -856,3 +860,110 @@ def renumber_roi_number(sequence):
+                    polygon_list.append(sub_polygon)
+            polygon = MultiPolygon(polygon_list)
+            roi_contour_sequence_geometry[slice_id] = polygon
+    return roi_contour_sequence_geometry
+
+
+def manipulate_rois(first_sequence_geometry, second_sequence_geometry, image_uids, operation):
+    """
+    Compute the intersection of two ROIs
+    :param first_sequence_geometry: The geometry dictionary of the first ROI
+    :param second_sequence_geometry: The geometry dictionary of the second ROI
+    :param image_uids: The list of all image uids
+    :param operation: A string of either "INTERSECTION", "UNION", or "DIFFERENCE"
+    :return: A dictionary with key-value pair {slice-id: Polygon/Multipolygon Object}
+    """
+    if operation not in ["INTERSECTION", "UNION", "DIFFERENCE"]:
+        raise ValueError("Operation must be either INTERSECTION/UNION/DIFFERENCE")
+
+    result_geometry_dict = {}
+    for slice_id in image_uids:
+        first_geometry = first_sequence_geometry.get(slice_id)
+        second_geometry = second_sequence_geometry.get(slice_id)
+        first_geometry = make_valid(first_geometry) if first_geometry else first_geometry
+        second_geometry = make_valid(second_geometry) if second_geometry else second_geometry
+
+        if not first_geometry and not second_geometry:
+            continue
+
+        if operation == "INTERSECTION" and first_geometry and second_geometry:
+            result_geometry = first_geometry.intersection(second_geometry)
+        elif operation == "UNION":
+            if not first_geometry and second_geometry:
+                result_geometry = second_geometry
+            elif not second_geometry and first_geometry:
+                result_geometry = first_geometry
+            else:
+                result_geometry = first_geometry.union(second_geometry)
+        elif operation == "DIFFERENCE":
+            if first_geometry and not second_geometry:
+                result_geometry = first_geometry
+            elif first_geometry and second_geometry:
+                result_geometry = first_geometry.difference(second_geometry)
+            else:
+                continue
+        else:
+            continue
+
+        result_geometry_dict[slice_id] = result_geometry
+    return result_geometry_dict
+
+
+def geometry_to_roi(roi_sequence_geometry):
+    """
+    Convert the geometry object in each image slice to ROI contour data
+    :param roi_sequence_geometry: A geometry dictionary
+    :return: A dictionary with key-value pair {slice-id: contour sequence}
+    """
+    roi_contour_sequence = {}
+    for slice_id, geometry in roi_sequence_geometry.items():
+        contour_sequence = []
+        geometry_type = geometry.geom_type
+        if not geometry.is_empty:
+            if geometry_type == 'Polygon':
+                contour_data = []
+                for coord in geometry.exterior.coords:
+                    contour_data.append(list(map(int, coord)))
```

Comment:
put this in a function `convert_polygon_to_roi()`

Reply:
This has been reduced to one line of code so no need to separate into another function.